internal/service/arcregionswitch/plan.go

```diff
@@ -0,0 +1,1373 @@
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func FindPlanByARN(ctx context.Context, conn *arcregionswitch.Client, arn string) (*awstypes.Plan, error) {
+	input := arcregionswitch.GetPlanInput{
+		Arn: aws.String(arn),
+	}
+
+	output, err := conn.GetPlan(ctx, &input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil || output.Plan == nil {
+		return nil, tfresource.NewEmptyResultError(input)
+	}
+
+	return output.Plan, nil
+}
+
+func ListTags(ctx context.Context, conn *arcregionswitch.Client, identifier string) (tftags.KeyValueTags, error) {
+	input := arcregionswitch.ListTagsForResourceInput{
+		Arn: aws.String(identifier),
+	}
+
+	output, err := conn.ListTagsForResource(ctx, &input)
+
+	if err != nil {
+		return tftags.New(ctx, nil), err
+	}
+
+	return tftags.New(ctx, output.ResourceTags), nil
+}
+
+func UpdateTags(ctx context.Context, conn *arcregionswitch.Client, identifier string, oldTagsMap, newTagsMap any) error {
+	oldTags := tftags.New(ctx, oldTagsMap)
+	newTags := tftags.New(ctx, newTagsMap)
+
+	if removedTags := oldTags.Removed(newTags); len(removedTags) > 0 {
+		input := arcregionswitch.UntagResourceInput{
+			Arn:             aws.String(identifier),
+			ResourceTagKeys: removedTags.Keys(),
+		}
+
+		_, err := conn.UntagResource(ctx, &input)
+
+		if err != nil {
+			return fmt.Errorf("untagging resource (%s): %w", identifier, err)
+		}
+	}
+
+	if updatedTags := oldTags.Updated(newTags); len(updatedTags) > 0 {
+		input := arcregionswitch.TagResourceInput{
+			Arn:  aws.String(identifier),
+			Tags: updatedTags.Map(),
+		}
+
+		_, err := conn.TagResource(ctx, &input)
+
+		if err != nil {
+			return fmt.Errorf("tagging resource (%s): %w", identifier, err)
+		}
+	}
+
+	return nil
+}
+
+// @FrameworkResource("aws_arcregionswitch_plan", name="Plan")
+// @Tags(identifierAttribute="arn")
+func newResourcePlan(context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourcePlan{}
+	return r, nil
+}
+
+type resourcePlan struct {
+	framework.ResourceWithConfigure
+}
+
+func (r *resourcePlan) ValidateModel(ctx context.Context, schema *fwschema.Schema) fwdiag.Diagnostics {
+	var diags fwdiag.Diagnostics
+	// Basic validation is handled by the schema validators
+	return diags
+}
+
+type resourcePlanModel struct {
+	ARN                          types.String `tfsdk:"arn"`
+	ID                           types.String `tfsdk:"id"`
+	Name                         types.String `tfsdk:"name"`
+	ExecutionRole                types.String `tfsdk:"execution_role"`
+	RecoveryApproach             types.String `tfsdk:"recovery_approach"`
+	Regions                      types.List   `tfsdk:"regions"`
+	Description                  types.String `tfsdk:"description"`
+	PrimaryRegion                types.String `tfsdk:"primary_region"`
+	RecoveryTimeObjectiveMinutes types.Int64  `tfsdk:"recovery_time_objective_minutes"`
+	AssociatedAlarms             types.Set    `tfsdk:"associated_alarms"`
```

Comment:
Example of the corresponding type definition on the data structure to match the `CustomType` argument in the schema definition. These need to be matching pairs for AutoFlex to function properly.

```suggestion
	AssociatedAlarms             fwtypes.SetNestedObjectValueOf[associatedAlarmModel]    `tfsdk:"associated_alarms"`
```